### DISCUSSION TITLE
Location of C API

### DISCUSSION THREAD

**Discussion by vincent-grosbois** [oneapi-src/oneDAL#1407] [Q&A] [answered]
Hello
After browsing the code and documentation for this project, I'm unable to find a C API / headers for oneDAL, does it exists?
It seems that these C bindings are mentionned on the Intel website, but I can't find them.
Otherwise, if they don't exist, how does daal4py interacts with the native OneDal code?

thanks !

**Comment by rlnx** [comment, marked as answer]
Hi, native part of oneDAL is C++ library, there is no C API. All C++ APIs/implementations are located in [<root>/cpp](https://github.com/oneapi-src/oneDAL/tree/master/cpp) directory. Currently there are two types of supported C++ APIs:
- CPU only API, which is called `daal` by historical reasons;
- New API called `oneapi`, which as part of [oneAPI open specification](https://www.oneapi.com/) and can run on various hardware including CPU, GPU, ...

At the moment daal4py is based on CPU only API, header files [can be found here](https://github.com/oneapi-src/oneDAL/tree/master/cpp/daal/include). 

All Python bindings are based on Cython and part of [daal4py repository](https://github.com/IntelPython/daal4py).

**Comment by vincent-grosbois** [comment]
Thanks for you answer! for some reason I thought that the python binding could only be done through pure C and not C++
